src/collision.cpp

```diff
@@ -249,10 +249,12 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 	} else {
 		time_notification_done = false;
 	}
+
+	v3f newpos_f = *pos_f + (*speed_f + accel_f * 0.5f * dtime) * dtime;
 	*speed_f += accel_f * dtime;
 
-	// If there is no speed, there are no collisions
-	if (speed_f->getLength() == 0)
+	// If the object is static, there are no collisions
+	if (newpos_f == *pos_f)
```

Comment:
general question (not for this PR): shouldn't this use a comparison with some margin of tolerance?
or would that introduce a bug where it might miss collisions?

Reply:
All movement happening at a speed below the margin would be completely ignored and would cause entities to miss collisions, yes.